dm/worker/subtask.go

```diff
@@ -600,10 +602,24 @@ func (st *SubTask) unitTransWaitCondition(subTaskCtx context.Context) error {
 			if err != nil {
 				return terror.WithClass(err, terror.ClassDMWorker)
 			}
+			if st.cfg.EnableGTID {
+				gset1, err := gtid.ParserGTID(st.cfg.Flavor, loadStatus.MetaBinlogGTID)
+				if err != nil {
+					st.l.Error("fail to parse gtid", log.ShortError(err))
+				}
+				gset2, err := gtid.ParserGTID(st.cfg.Flavor, relayStatus.RelayBinlogGtid)
+				if err != nil {
+					st.l.Error("fail to parse gtid", log.ShortError(err))
+				}
+				rc, ok := binlog.CompareGTID(gset1, gset2)
```

Comment:
maybe we should handle not `ok` case instead of dead loop

Reply:
If not ok, will compare pos below(L619).

Reply:
Generate nil previous gtid event. Like `reset master`

Reply:
I didn't learn GTID much, will there be not comparable `gset1` `gset2`, and is falling to binlog position compare a good way to solve?

Reply:
If gset1 can't compare gset2, there must be some error occured and syncer also can't sync from dump_data/metadata. I think both deed loop or report a error is ok?

Reply:
I prefer to return the error explicitly and fast failure.